tests/test_utils/tests.py

```diff
@@ -2186,6 +2186,8 @@ def test_allowed_database_copy_queries(self):
         finally:
             new_connection.validate_thread_sharing()
             new_connection._close()
+            if new_connection.is_pool:
+                new_connection.close_pool()
```

Comment:
I'm not really sure why this needs a `close_pool` but `test_allowed_threaded_database_queries` does not 
What happens when you remove this?

Reply:
In the `test_allowed_database_copy_queries` test, it's necessary to explicitly close the pool because it copies a connection with the `dynamic_connection` alias. Due to this new alias, a new pool is created with the key `(dynamic_connection, user)` , and this pool is never closed because `new_connection._close()` only releases the connection back to the pool, but never closes the connection. As a result, when dropping the test user at the end of the test suite run, an error is raised because the test user is already connected due to this test.

In the case of `test_allowed_threaded_database_queries`, the alias is `default`, so it doesn't create a new pool; instead, it reuses the existing pool `(default, user)` and gets closed in the `_destroy_test_db()` call.

So if you delete this line, at the end of the test suite run, it will raise **cannot drop a user which is already connected** during the `_destroy_test_user()` call.

Reply:
Thank you for the context! Using a connection pool config for postgres also has an error when destroying the test db for this test

@apollo13 I feel like some of the test errors when running a connection pool config for postgres where ignored as "expected errors". Do you think it's worth adding a `close_pool` call to `test_allowed_database_copy_queries`?

If we don't want to add a `is_pool` property, we could do something like
```python
    if hasattr(new_connection, "close_pool"):
        new_connection.close_pool()
```

@suraj-ora-2020 note that `is_pool` is currently `False` for postgres connection pools - but I'm still not 100% convinced we want to add this property

Reply:
@sarahboyce , I agree that `is_pool` is currently `False` for postgres connection pools and other database backends. However, I think the pooling functionality will be extended to other backends in the future, including sqlite and mysql. In the base layer, we can set `is_pool` as False, and in the backend-specific layer, we can add the following code:

```python
@property
def is_pool(self):
    return self.settings_dict["OPTIONS"].get("pool", False)
```

This code can be added to postgres and the other backends when they extend this functionality.
If you have any other suggestions or if you're not convinced, feel free to let me know.

Reply:
I have updated the code according to the suggestion. This will work for both Oracle and PostgreSQL and will resolve the issue we are facing when dropping the user in the case of connection pooling for both backends.
`if hasattr(new_connection, "close_pool"):
        new_connection.close_pool()
`